src/foam/nanos/auth/PreventDuplicateEmailLoginService.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+   package: 'foam.nanos.auth',
+   name: 'PreventDuplicateEmailLoginService',
+   extends: 'foam.nanos.auth.ProxyUserLocatorService',
+   flags: ['java'],
+
+  documentation: 'Prevent users from logging in with duplicate email',
+
+  javaImports: [
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.dao.Sink',
+    'foam.nanos.auth.AuthenticationException',
+    'foam.nanos.auth.User',
+    'foam.nanos.theme.Theme',
+
+    'java.util.List',
+
+    'static foam.mlang.MLang.*'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'superSpid',
+      documentation: 'Set spid of a theme that is accessible to all users',
+      value: "",
+    }
+  ],
```

Comment:
I dont think you need it here

Reply:
right, we don't use this here. I removed it.